### PR TITLE
Grid Templates

### PR DESCRIPTION
**Pull request by gdomanguera**
**Grid Templates Specifications:**
•	A template is used to define a grid configuration.
•	This template will contain all the configuration that we currently do for a Grid.
•	The user can add/modify/delete a Grid Template
•	The user can create a Grid from a Grid Template
•	When we create a test, we create it from the Grid as we currently do.

**Sample use case:**  
•	Run a test every 2 weeks that are identical.  User can select the same Grid Template and create a grid from that each time they run this test.

**Note:**
•	This will require a new table in cockroachdb `grid_template` which is added in the `tables.txt` script

### CODE REVIEW

File: services/swarmhub/src/swarmhub/api/gridtemplate.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	}
+
+	gridTemplate, err = db.CreateGridTemplate(gridTemplate)
+	if err != nil {
+		message := fmt.Sprintf("error creating grid template: " + err.Error())
+		http.Error(w, message, http.StatusInternalServerError)
+		return
+	}
+
+	json, err := json.Marshal(gridTemplate)
```
Comment (kmhagan) [inline services/swarmhub/src/swarmhub/api/gridtemplate.go]: Can you rename the Marshal response from "json" to "b" or "jsonData"? Shadowing the json package could lead to confusion down the line

File: services/swarmhub/src/swarmhub/api/gridtemplate.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if len(gridTemplates) == 0 {
+		http.Error(w, "No content", http.StatusNoContent)
+		return
+	}
+
+	json, err := json.Marshal(gridTemplates)
```
Comment (kmhagan) [inline services/swarmhub/src/swarmhub/api/gridtemplate.go]: same here

File: services/swarmhub/src/swarmhub/api/gridtemplate.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+func GetGridTemplateById(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	id := ps.ByName("id")
+
+	gridTemplate, err := db.GetGridTemplateById(id)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	json, err := json.Marshal(gridTemplate)
```
Comment (kmhagan) [inline services/swarmhub/src/swarmhub/api/gridtemplate.go]: and here

**Review by kmhagan — CHANGES_REQUESTED** [review]
Looks good! If you could make the small variable name changes I put in the comments I think this is ready to be merged

**Comment by gdomanguera** [comment]
@kmhagan variable updated and renamed to `jsonData`

**Pull request merged**